### ISSUE TITLE
Full Site Editing: Deleting block settings padding doesn't reset value

### ISSUE THREAD

**Issue by jeyip** [WordPress/gutenberg#36794]
### Description

When I want to reset padding in block settings, I can't simply delete the current value. I have to explicitly set the value to 0 to remove padding.

### Step-by-step reproduction instructions

1. Enable full site editing by activating the Gutenberg plugin and an FSE enabled theme (Ex. Quadrat)
2. Navigate to the site editor
3. Add a row block
4. Add two items into the row block
5. Open the row block settings sidebar
6. Change the content justification to `space between`
7. Unselect the row block
8. Reselect it and verify that the content jumps

### Screenshots, screen recording, code snippet

![2021-11-23 15 38 47](https://user-images.githubusercontent.com/5414230/143146893-87a3d164-2854-423e-9ba2-29358486dc6a.gif)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by carolinan** [comment]
Removing the value resets to the theme or block default, not 0.

**Comment by Andrew-Starr** [comment]
The value can be reset by clicking the three dots to the right of 'Dimensions' and then selecting the 'Reset All' option.

It's not immediately clear how to do this and I struggled for a while until I found it.

**Comment by jeyip** [comment]
> Removing the value resets to the theme or block default, not 0.

Thanks for chiming in @carolinan That doesn't appear to be the case for me. In gif above, I set the padding to 34px, and when I delete the values, the padding in the block remains at 34px.

> The value can be reset by clicking the three dots to the right of 'Dimensions' and then selecting the 'Reset All' option. It's not immediately clear how to do this and I struggled for a while until I found it.

Thanks Andrew! That's helpful 🙏. I still think there's a concern for me regarding how intuitive the existing reset option is. I've seen that deleting input content on other platforms will reset the value automatically (although I can't seem to remember which ones anymore). Curious if either Carolina or you have more thoughts on this.

**Comment by carolinan** [comment]
But don't you need to save those changes for them to apply?

**Comment by carolinan** [comment]
I understand, I apologies for not paying close enough attention to the gif :)

**Comment by Andrew-Starr** [comment]
> Thanks Andrew! That's helpful 🙏. I still think there's a concern for me regarding how intuitive the existing reset option is. I've seen that deleting input content on other platforms will reset the value automatically (although I can't seem to remember which ones anymore). Curious if either Carolina or you have more thoughts on this.

I agree. My thoughts are that deleting the input should reset the padding. That would be the most intuitive.

Also worth noting that _**if**_ the padding is defined as 34px for the group/row block - either in theme.json or user defined in the site editor - that this value does not appear by default in the input control and is still empty.
So when resetting, the input goes back to empty but the block as displayed in the editor and frontend will still appear with the 34px padding.

**Comment by carolinan** [comment]
I believe there is an issue for the part where the "adjusted default value" is not showing in the control, but finding issues here among 3.6K is an issue.
